packages/java-edition/src/mcfunction/node/argument.ts

```diff
@@ -119,8 +120,21 @@ export namespace EntitySelectorArgumentsNode {
 		)
 	}
 }
-export const EntitySelectorVariables = ['a', 'e', 'p', 'r', 's'] as const
+export const EntitySelectorVariables = ['a', 'e', 'p', 'r', 's', 'n']
 export type EntitySelectorVariable = typeof EntitySelectorVariables[number]
+export function removeUnavailableSelectorVariables(ctx: core.ContextBase) {
+	const release = ctx.project['loadedVersion'] as
+		| ReleaseVersion
+		| undefined
+	// Remove @n if the version is lower than 1.21
+	if (
+		EntitySelectorVariables.includes('n') && release &&
+		ReleaseVersion.cmp(release, '1.21') < 0
+	) {
+		EntitySelectorVariables.pop()
+		EntitySelectorAtVariables.pop()
```

Comment:
This is really ugly and probably a bad idea to modify a global constant. Instead consider adding a `filterEntitySelectorVariables(ctx): string[]` pure function (or similarly named). This would also break in the future if we end up supporting multiple versions or allow switching the loaded version without a full vscode reload.

Reply:
The problem is the completer uses the list for suggestions, and I couldn't figure out another way to modify the suggestions without editing the list. Any suggestions @misode?